packages/jest-runner/src/test_worker.js

```diff
@@ -12,12 +12,6 @@ import type {GlobalConfig, Path, ProjectConfig} from 'types/Config';
 import type {SerializableError, TestResult} from 'types/TestResult';
 import type {RawModuleMap} from 'types/HasteMap';
 
-// Make sure uncaught errors are logged before we exit.
-process.on('uncaughtException', err => {
```

Comment:
I have pushed a version that removes and restores the listeners from jasmine, which seems to work well.